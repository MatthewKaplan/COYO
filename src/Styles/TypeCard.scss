// @import 'Mixins';

// *,
// *:after,
// *:before {
//   box-sizing: border-box;
//   font-family: 'Raleway', sans-serif;
// }

// .imageContainer {
//   // width: 30%;
//   margin: 100px 30px;
// }

// .typeSmall {
//   // width: 100%;
//   // height: 500px;
//   // max-width: 400px;
//   // width: 400px;
  
//   // transition:all 0.3s ease;

//   // &&:hover {
//   //   -webkit-transform: scale(1.05);
//   //   -ms-transform: scale(1.05);
//   //   transform: scale(1.05);
//   //   box-shadow:5px 5px 5px hsl(204, 42%, 27%);
//   // }
// }

// .typeSmallImg {
//   width: 100%;
  
// }

// .yogaTypeCard {
//   border: 2px solid black;
//   border-radius: 5px;
//   box-shadow:  3px 2px 2px #284B63;
//   overflow-wrap: break-word;
//   min-height: 300px;
//   margin: 10px;
//   // max-width: 275px;
//   padding: 20px;
//   background: #eff0f2;
//   display: none;

// }



// .typeHeading {
//   font-weight: bold;
//   font-size: 22px;
//   letter-spacing: 3px;
// }

// .purpose {
//   font-size: 16px;
//   font-style: italic;
// }

// .beginner {
//   margin-bottom: 0px;
// }

