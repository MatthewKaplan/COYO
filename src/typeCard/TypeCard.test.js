import React from 'react';
import TypeCard from './TypeCard';
import { shallow } from 'enzyme';

const mockType = {
  name: "Hatha",
  typeId: 1,
  propsUsed: false,
  commonPoses: ["Vrikshasana",
  "Salabhasana",
  "Tadasana",
  "Uttanasana",
  "Adho Mukha Svanasana",
  "Setu Bandhasana"],
  beginnerFriendly: true,
  purpose: "To align and calm your body, mind, and spirit in preparation for meditation."
}

describe('TypeCard', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(
      <TypeCard name={mockType.name}
                purpose={mockType.purpose}
                beginnerFriendly={mockType.beginnerFriendly}
                propsUsed={mockType.propsUsed}
                commonPoses={mockType.commonPoses}
                />
    )
  });
  it('should render correctly with all the information passed', () => {
    expect(wrapper).toMatchSnapshot();
  });

  

});
