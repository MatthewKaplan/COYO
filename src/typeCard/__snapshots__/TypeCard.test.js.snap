// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeCard should render correctly with all the information passed 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TypeCard
    beginnerFriendly={true}
    commonPoses={
      Array [
        "Vrikshasana",
        "Salabhasana",
        "Tadasana",
        "Uttanasana",
        "Adho Mukha Svanasana",
        "Setu Bandhasana",
      ]
    }
    name="Hatha"
    propsUsed={false}
    purpose="To align and calm your body, mind, and spirit in preparation for meditation."
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4
          className="typeHeading"
        >
          Hatha
        </h4>,
        <h5
          className="purpose"
        >
          Purpose:
        </h5>,
        <h5
          className="purpose"
        >
          To align and calm your body, mind, and spirit in preparation for meditation.
        </h5>,
        <p
          className="beginner"
        >
          Difficulty:
        </p>,
        <p
          className="beginner"
        >
          Beginner friendly
        </p>,
        <p
          className="poses"
        >
          Poses: 
        </p>,
        <p
          className="poses"
        >
          Vrikshasana
          Salabhasana
          Tadasana
          Uttanasana
          Adho Mukha Svanasana
          Setu Bandhasana
        </p>,
        <input
          className="findStudioBtn"
          onClick={[Function]}
          type="submit"
          value="Find Studios"
        />,
      ],
      "className": "yogaTypeCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hatha",
          "className": "typeHeading",
        },
        "ref": null,
        "rendered": "Hatha",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Purpose:",
          "className": "purpose",
        },
        "ref": null,
        "rendered": "Purpose:",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "To align and calm your body, mind, and spirit in preparation for meditation.",
          "className": "purpose",
        },
        "ref": null,
        "rendered": "To align and calm your body, mind, and spirit in preparation for meditation.",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Difficulty:",
          "className": "beginner",
        },
        "ref": null,
        "rendered": "Difficulty:",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Beginner friendly",
          "className": "beginner",
        },
        "ref": null,
        "rendered": "Beginner friendly",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Poses: ",
          "className": "poses",
        },
        "ref": null,
        "rendered": "Poses: ",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Vrikshasana",
            "Salabhasana",
            "Tadasana",
            "Uttanasana",
            "Adho Mukha Svanasana",
            "Setu Bandhasana",
          ],
          "className": "poses",
        },
        "ref": null,
        "rendered": Array [
          "Vrikshasana",
          "Salabhasana",
          "Tadasana",
          "Uttanasana",
          "Adho Mukha Svanasana",
          "Setu Bandhasana",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "findStudioBtn",
          "id": undefined,
          "onClick": [Function],
          "type": "submit",
          "value": "Find Studios",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className="typeHeading"
          >
            Hatha
          </h4>,
          <h5
            className="purpose"
          >
            Purpose:
          </h5>,
          <h5
            className="purpose"
          >
            To align and calm your body, mind, and spirit in preparation for meditation.
          </h5>,
          <p
            className="beginner"
          >
            Difficulty:
          </p>,
          <p
            className="beginner"
          >
            Beginner friendly
          </p>,
          <p
            className="poses"
          >
            Poses: 
          </p>,
          <p
            className="poses"
          >
            Vrikshasana
            Salabhasana
            Tadasana
            Uttanasana
            Adho Mukha Svanasana
            Setu Bandhasana
          </p>,
          <input
            className="findStudioBtn"
            onClick={[Function]}
            type="submit"
            value="Find Studios"
          />,
        ],
        "className": "yogaTypeCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hatha",
            "className": "typeHeading",
          },
          "ref": null,
          "rendered": "Hatha",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Purpose:",
            "className": "purpose",
          },
          "ref": null,
          "rendered": "Purpose:",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "To align and calm your body, mind, and spirit in preparation for meditation.",
            "className": "purpose",
          },
          "ref": null,
          "rendered": "To align and calm your body, mind, and spirit in preparation for meditation.",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Difficulty:",
            "className": "beginner",
          },
          "ref": null,
          "rendered": "Difficulty:",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Beginner friendly",
            "className": "beginner",
          },
          "ref": null,
          "rendered": "Beginner friendly",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Poses: ",
            "className": "poses",
          },
          "ref": null,
          "rendered": "Poses: ",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Vrikshasana",
              "Salabhasana",
              "Tadasana",
              "Uttanasana",
              "Adho Mukha Svanasana",
              "Setu Bandhasana",
            ],
            "className": "poses",
          },
          "ref": null,
          "rendered": Array [
            "Vrikshasana",
            "Salabhasana",
            "Tadasana",
            "Uttanasana",
            "Adho Mukha Svanasana",
            "Setu Bandhasana",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "findStudioBtn",
            "id": undefined,
            "onClick": [Function],
            "type": "submit",
            "value": "Find Studios",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
